--!strict

export type Paths = {
	ManualTree: string,
	ManualTreeTarget: string,
	TreeLeaves: string,
	ManualRock: string,
	ManualRockTarget: string,
	NPCPurchasePad: string,
	NPCPurchasePadTarget: string,
	StorageUpgradeButton: string,
	StorageUpgradeButtonTarget: string,
	Machine: string,
	MachineRuntimePieces: string,
	MachineStorageOutput: string,
	Conveyor1: string,
	Conveyor1Interaction: string,
	Upgrader1: string,
	Upgrader1Interaction: string,
	CoalProcessor: string,
	CoalProcessorInteraction: string,
	Conveyor2: string,
	Conveyor2Interaction: string,
	Upgrader2: string,
	Upgrader2Interaction: string,
	CollectionPoint: string,
	CollectionPointInteraction: string,
	SellTouchPad: string,
	SellBillboard: string,
	SellBillboardLabel: string,
	SellScreenLabel: string,
	WorkerHomes: string,
	WorkerApproachPoints: string,
	TreeApproach: string,
	RockApproach: string,
	StorageApproach: string,
	StorageDisplayAnchor: string,
	StorageDropTarget: string,
	PlayerSpawn: string,
}

return {}
