--!strict

export type Paths = {
	Ground: string,
	NPCPurchasePad: string,
	NPCPurchasePadTarget: string,
	StorageUpgradeButton: string,
	StorageUpgradeButtonTarget: string,
	Machine: string,
	MachineRuntimePieces: string,
	MachineStorageOutput: string,
	Conveyor1: string,
	Conveyor1Interaction: string,
	Upgrader1: string,
	Upgrader1Interaction: string,
	CoalProcessor: string,
	CoalProcessorInteraction: string,
	Conveyor2: string,
	Conveyor2Interaction: string,
	Upgrader2: string,
	Upgrader2Interaction: string,
	CollectionPoint: string,
	CollectionPointInteraction: string,
	SellTouchPad: string,
	SellScreen: string,
	WorkerHomes: string,
	WorkerApproachPoints: string,
	StorageApproach: string,
	CollectionPointApproach: string,
	StorageDisplayAnchor: string,
	StorageDropTarget: string,
	PlayerSpawn: string,
	Drops: string,
	BuildGridBounds: string,
	BuildGridBlockers: string,
	BuildGridPlacedItems: string,
	BuildGridRuntimeItems: string,
}

return {}
